### PR TITLE
#10824: Onboarding telemetry

### PR DESCRIPTION
**Pull request by eliserichards**
For #10824 

Events: 
- [x]  OnboardingManualSignIn 
- [x]  OnboardingAutomaticSignIn
- [x]  OnboardingThemePicker
   - dark, light, automatic (follow device settings)
   - default: automatic
- [x]  OnboardingTrackingProtection
   - strict, standard
   - default: ON, standard
- [x]  OnboardingPrivateBrowsing
- [x]  OnboardingPrivacyNotice
- [x]  OnboardingFinish
- [x]  OnboardingToolbarPositionPicker 
   - top, bottom
   - default: bottom
- [x]  OnboardingWhatsNew

**This PR is covering:**
  * User toggled any of the onboarding toggles

**Already exists in preference metrics** 
  * Track final position of these toggles through our metrics ping 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/metrics.yaml [hunk 1]
```diff
@@ -122,7 +122,53 @@ events:
     notification_emails:
       - fenix-core@mozilla.com
     expires: "2020-09-01"
-  preference_toggled:
+  whats_new_tapped:
```
Comment (eliserichards) [inline app/metrics.yaml:125]: This ordering got all messed up. I pulled `preference_toggled` down into it's own section (per discussion with @sblatz and Marissa) and nested all onboarding clicks within that

**Comment by liuche** [comment]
@eliserichards can you include a data review form please?

**Comment by eliserichards** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- What percentage of users interact with onboarding cards by changing a toggle or tapping on a link (any of them)?
- What percentage of users interact with each individual card (which cards resonate more with users and make them want to explore/take action/change anything?)
- What percentage of users who change a preference (e.g. ETP) do it during onboarding vs in settings later on?
- Are users who interact with onboarding more likely to come back in the 7 days following install?

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- We would like to better understand the onboarding flow in order to streamline the experience for users, and to ensure that the initial interaction with the app is useful and doesn't drive people away.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* There is not currently an alternative to measure this. 

4) **Can current instrumentation answer these questions?**

* No, there are no currently metrics around the onboarding actions.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 2.

6) **How long will this data be collected?**

* Until 09/01/2020

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Glean, Amplitude and with mobile teams.

File: app/metrics.yaml [hunk 2]
```diff
@@ -205,6 +205,128 @@ events:
...
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/10824
+    notification_emails:
+      - fenix-core@mozilla.com
+      - erichards@mozilla.com
+    expires: "2020-09-01"
+  pref_toggled_toolbar_position:
+    type: event
+    description:
+      The toolbar position preference was chosen from the onboarding card.
```
Comment (sblatz) [inline app/metrics.yaml:260]: Let's make sure to include the default state for all of these prefs getting toggled 😄

File: app/metrics.yaml [hunk 3]
```diff
@@ -205,6 +205,128 @@ events:
...
+      - erichards@mozilla.com
+    expires: "2020-09-01"
+  privacy_notice:
+    type: event
+    description:
+      The onboarding privacy notice card was tapped.
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/10824
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/10824
```
Comment (sblatz) [inline app/metrics.yaml:240]: You should link directly to this PR for the data review link.

File: app/metrics.yaml [hunk 4]
```diff
@@ -205,6 +205,131 @@ events:
...
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/11867
+    notification_emails:
+      - fenix-core@mozilla.com
+      - erichards@mozilla.com
+    expires: "2020-09-01"
+  pref_toggled_private_browsing:
+    type: event
+    description:
+      The private browsing preference was selected from the onboarding card.
```
Comment (liuche) [inline app/metrics.yaml:248]: Does this automatically trigger the PB mode being saved to the metrics ping for private browsing setting? https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md#metrics-1

Comment (eliserichards) [reply app/metrics.yaml:248]: Yes it does!

**Pull request merged**